Login&register

MBabiar opened the pull request.

MBabiar committed:
feat(Created Login & Register Pages):
- Created AuthLayout
- Created Login & Register Pages
- Changed primary color

MBabiar committed:
feat(Improved login/register pages):
- Added cssAddon: true to quasar.config
- Changed grey-1 variable
- Improved AuthLayout
- Improved Login & Register Pages
- Added Login and Register Validations
- Added Login and Register redirect on successful form submit
- Added IndexPage to route

samuelcseto commented on src/pages/auth/LoginPage.vue:
```diff
@@ -0,0 +1,67 @@
+        v-model="loginForm.password"
+        label="Password"
+        :type="passwordVisibility ? 'password' : 'text'"
+        :rules="[(val: any) => !!val || 'Password is required', (val: any) => validatePassword(val) || 'Password must be at least 8 characters']"
```
Consider renaming this function as it only checks for password length

samuelcseto commented on src/pages/auth/LoginPage.vue:
```diff
@@ -0,0 +1,67 @@
+  })
+
+  function validateEmail(email: string): boolean {
+    return /[a-z0-9]+@[a-z]+\.[a-z]{2,3}/.test(email)
```
Validating e-mail addresses by regex is considered not good practice as TLDs are being constantly expanded and this will not allow TLDs with more than 3 characters.

samuelcseto commented on src/pages/auth/LoginPage.vue:
```diff
@@ -0,0 +1,67 @@
+  }
+
+  function validatePassword(password: string): boolean {
+    return password.length >= 8
```
Consider adding more validation logic, such as
- contains uppercase letter
- contains number

samuelcseto commented on src/pages/auth/LoginPage.vue:
```diff
@@ -0,0 +1,67 @@
+  }
+
+  const onLogin = () => {
+    console.log('Login:', loginForm.value)
```
Console.log commands should be purged/removed before commiting a change

samuelcseto commented on src/pages/auth/RegisterPage.vue:
```diff
@@ -0,0 +1,102 @@
+    name: 'RegisterPage'
+  })
+
+  const passwordVisibility = ref(true)
```
Is the password visible by default? If yes, please change. If not, rephrase the naming so it's not misleading.

MBabiar replied:
All above fixed

MBabiar committed:
feat:
- Added validator library for email validation
- Updated email and password validation
- Removed console logs
